api/api.go

```diff
@@ -695,11 +705,31 @@ func matchFilterLabels(matchers []*labels.Matcher, sms map[string]string) bool {
 	for _, m := range matchers {
 		v, prs := sms[m.Name]
 		switch m.Type {
-		case labels.MatchNotEqual, labels.MatchNotRegexp:
+		case labels.MatchNotRegexp:
+			if string(m.Value) == "^(?:)$" && prs {
```

Comment:
The `string(m.Value) == "^(?:)$"` test seems brittle to me. Also this will break if we update the vendored `github.com/prometheus/prometheus/pkg/labels` package because in the latest version, the original `m.Value` is kept (no anchoring added, see [this](https://github.com/prometheus/prometheus/blob/5b962c57489f6995f24c56fc4ac8b97faec753fd/pkg/labels/matcher.go#L55-L69)). Maybe updating `github.com/prometheus/prometheus/pkg/labels` and matching againt `""` then?

Reply:
yeah, 100%. Good to hear that the latest version doesn't change `m.Value`, will update that and simplify this.